src/Polly.Core/README.md

```diff
@@ -0,0 +1,204 @@
+    
+    // omitted for simplicity
+}
+```
+
+The `ResilienceContext` is defined as:
+
+``` csharp
+public sealed class ResilienceContext
+{
+    public CancellationToken CancellationToken { get; set; }
+
+    public bool IsSynchronous { get; }
+
+    public bool IsVoid { get; }
+
+    public bool ContinueOnCapturedContext { get; }
+
+    // omitted for simplicity
+}
+```
+
+The `ResilienceStrategy` unifies the 4 different policies used now in Polly. User actions are executed under a single API. The are many methods
+exposed on this class that cover different scenarios:
+
+- Synchronous void methods.
+- Synchronous methods with result.
+- Asynchronous void methods.
+- Asynchronous methods with result.
+
+For example, synchronous `Execute` method is implemented as:
+
+``` csharp
+public void Execute(Action execute)
+{
+    var context = ResilienceContext.Get();
+    context.IsSynchronous = true;
+    context.IsVoid = true;
+
+    try
+    {
+        strategy.ExecuteCoreAsync(static (context, state) =>
+        {
+            state();
+            return new ValueTask<VoidResult>(VoidResult.Instance);
+        }, 
+        context, 
+        execute).GetAwaiter().GetResult();
+    }
+    finally
+    {
+        ResilienceContext.Return(context);
+    }
+}
+```
+
+In the preceding example:
+
+- We rent `ResilienceContext` from pool.
+- We store the information about the execution mode by setting the `IsSynchronous` and `IsVoid` properties to the context.
+- We pass the user delegate, and use the `State` to avoid closure allocation.
+- We block the execution.
+- We return `ResilienceContext` to the pool.
+
+Underlying implementation decides how to execute this delegate by reading the `ResilienceContext`:
+
+``` csharp
+internal class DelayStrategy : DelegatingResilienceStrategy
+{
+    protected override async ValueTask<T> ExecuteCoreAsync<T, TState>(Func<ResilienceContext, TState, ValueTask<T>> execution, ResilienceContext context, TState state)
+    {
+        if (context.IsSynchronous)
+        {
+            Thread.Sleep(1000);
+        }
+        else
+        {
+            await Task.Delay(1000);
+        }
+
+        return await execution(context, state);
+    }
+}
+```
+
+In the preceding example:
+
+- For synchronous execution we are using `Thread.Sleep`.
+- For asynchronous execution we are using `Task.Delay`.
+
+This way, the responsibility of how to execute method is lifted from the user and instead passed to the policy. User cares only about the `ResilienceStrategy` class. User uses only a single strategy to execute all scenarios. Previously, user had to decide whether to use sync vs async, typed vs non-typed policies.
+
+The life of extensibility author is also simplified as they only maintain one implementation of strategy instead of multiple ones. See the duplications in [`Polly.Retry`](https://github.com/App-vNext/Polly/tree/main/src/Polly/Retry).
```

Comment:
```suggestion
The life of an extensibility author is also simplified as they only need to maintain one implementation of a strategy instead of multiple ones. See the duplications in [`Polly.Retry`](https://github.com/App-vNext/Polly/tree/main/src/Polly/Retry).
```